diff --git a/app/Model/Server.php b/app/Model/Server.php
index a8214a1b9..0ce5cabae 100644
--- a/app/Model/Server.php
+++ b/app/Model/Server.php
@@ -1536,7 +1536,7 @@ class Server extends AppModel
             return true;
         }
         if (is_executable($value)) {
-            if (finfo_file($finfo, $value) == "application/x-executable" || finfo_file($finfo, $value) == "application/x-sharedlib") {
+            if (finfo_file($finfo, $value) == "application/x-executable" || finfo_file($finfo, $value) == "application/x-pie-executable" || finfo_file($finfo, $value) == "application/x-sharedlib") {
                 finfo_close($finfo);
                 return true;
             } else {
